modules/telestion-application/src/main/java/de/wuespace/telestion/application/loader/TelestionLoader.java

```diff
@@ -0,0 +1,323 @@
+package de.wuespace.telestion.application.loader;
+
+import de.wuespace.telestion.application.deployment.Deployment;
+import de.wuespace.telestion.application.launcher.ConfigDeploymentLauncher;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.ParameterizedType;
+import java.util.Objects;
+
+/**
+ * An abstract reference implementation of a {@link Loader}.
+ * <p>
+ * Use this as a starting point for your own loader.
+ *
+ * @param <T> the type of the loader configuration
+ * @author Ludwig Richter (@fussel178)
+ * @see Loader
+ * @see LoaderConfiguration
+ * @see ConfigDeploymentLauncher
+ */
+public abstract class TelestionLoader<T extends LoaderConfiguration> implements Loader<T> {
+
+	/**
+	 * The default loader configuration in the raw format.e.printStackTrace();
+	 */
+	private JsonObject genericDefaultConfig = new JsonObject();
+
+	/**
+	 * The default loader configuration in the type-safe format.
+	 */
+	private T defaultConfig;
+
+	/**
+	 * The loader configuration in the raw format.
+	 */
+	private JsonObject genericConfig = new JsonObject();
+
+	/**
+	 * The loader configuration in the type-safe format.
+	 */
+	private T config;
+
+	/**
+	 * The reference to the {@link ConfigDeploymentLauncher Launcher} instance which deployed this loader.
+	 */
+	protected ConfigDeploymentLauncher<? extends JsonObject> launcher;
+
+	/**
+	 * Map a generic JSON object to the Configuration type.<p>
+	 * Returns <code>null</code> when no type via {@link #getConfigType()} is given.
+	 *
+	 * @param object the generic JSON object to map
+	 * @return the JSON object in the Configuration type format
+	 */
+	private T mapToConfiguration(JsonObject object) {
+		var type = getConfigType();
+		return Objects.isNull(type) ? null : object.mapTo(type);
+	}
+
+	/**
+	 * Merges in the raw configuration into the default configuration
+	 * and updates both raw and type-safe configuration.
+	 *
+	 * @param rawConfig the raw configuration to merge into the default configuration
+	 */
+	private void mergeInConfiguration(JsonObject rawConfig) {
+		this.genericConfig = this.genericDefaultConfig.mergeIn(rawConfig);
+		this.config = mapToConfiguration(this.genericConfig);
+	}
+
+	/**
+	 * The default logger instance.
+	 */
+	protected final Logger logger = LoggerFactory.getLogger(getClass());
+
+	/**
+	 * Get the Configuration Class type from the inheriting class.
+	 *
+	 * @return the Configuration Class type
+	 */
+	@SuppressWarnings("unchecked")
+	protected Class<T> getConfigType() {
+		try {
+			String className = ((ParameterizedType) getClass().getGenericSuperclass()).getActualTypeArguments()[0].getTypeName();
+			Class<?> clazz = Class.forName(className);
+			//noinspection unchecked
+			return (Class<T>) clazz;
+		} catch (Exception e) {
+			logger.warn("Cannot get Class type from generic: {}", e.getMessage());
```

Comment:
Not a big fan of this warning message, tbh

Reply:
That's interesting, we have this exact same warning message in the `TelestionVerticle` class :thinking: 

https://github.com/wuespace/telestion-core/blob/276285c1ba23d6f86ed22813384bfc8d7dd19ec4/modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/TelestionVerticle.java#L62

Reply:
I think, we should change that, too :see_no_evil:

Reply:
Done in [`d2e327a` (#502)](https://github.com/wuespace/telestion-core/pull/502/commits/d2e327a5fbf8031f806fc5f24c7786e58607d26d)